src/components/TableHeaderCell/TableHeaderCell.module.css

```diff
@@ -0,0 +1,34 @@
+@import '../../design-tokens/mixins.css';
+
+/*------------------------------------*\
+    # TABLE HEADER CELL
+\*------------------------------------*/
+
+/**
+ * Table header cell
+ */
+:where(.table-header-cell) {
```

Comment:
Why do we need the `:where()` here?

Reply:
All the `where()` pseudoclasses raised as part of this pr were due to the subcomponents styles not being overriden with utility classes, i.e. https://github.com/chanzuckerberg/edu-design-system/blob/f063dae0ba4d5b41c927588439bee4bac7b39809/src/components/Table/Table.stories.tsx#L172 wouldn't work without this pseudoclass.

I'll add comments in the files explaining the need to lower specificity